lib/web_ui/lib/src/engine/compositor/image.dart

```diff
@@ -48,7 +48,23 @@ class CkAnimatedImage implements ui.Image {
   @override
   Future<ByteData> toByteData(
       {ui.ImageByteFormat format = ui.ImageByteFormat.rawRgba}) {
-    throw 'unimplemented';
+    Uint8List bytes;
+
+    if (format == ui.ImageByteFormat.rawRgba) {
+      final SkImageInfo imageInfo = SkImageInfo(
+        alphaType: canvasKit.AlphaType.Premul,
+        colorType: canvasKit.ColorType.RGBA_8888,
+        width: width,
+        height: height,
+      );
+      bytes = _skAnimatedImage.readPixels(imageInfo, 0, 0);
+    } else {
+      final SkData skData = _skAnimatedImage.encodeToData(); //defaults to PNG 100%
+      bytes = canvasKit.getSkDataBytes(skData);
+    }
+
+    final ByteData data = Uint8List.fromList(bytes).buffer.asByteData(0, bytes.length);
```

Comment:
I seem to recall that using the original buffer resulted in it going away by the time it's needed. I'll check it again with the current setup and either remove or comment.

Reply:
Yes. `getSkDataBytes()` returns a `SkData` and that seems not to worry about us needing the bytes later. We immediately run into a

```
Uncaught TypeError: Missing field
    at Object.toWireType (canvaskit.js:218)
    at SkImage$_readPixels [as _readPixels] (eval at Kc (canvaskit.js:155), <anonymous>:9:26)
    at SkImage.a.SkImage.readPixels (canvaskit.js:46)
    at JsObject.callMethod$2 (list_body.dart:176)
    at CkImage.toByteData$1$format (svg_path.dart:125)
```

If you think there is a better way to do it... Do you have some internal cloning mechanism that's cheaper and normally gets used in similar situations?

Reply:
Ah, makes sense. Yes, when the WASM memory gets resized any typed array attached to it gets detached. There is a way to fix this, but for now let's copy it out of the WASM memory. Please leave a comment explaining why this is necessary. We will revisit this code when we implement image decoding on the web-worker to avoid janking, and might optimize it at that time.